### PR TITLE
WFLY-20140 - Refactor WildFlyOpenTelemetryConfig

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-20140

Remove old constructor
Add Builder helper class
Update references

### CODE REVIEW

**Comment by bstansberry** [comment]
Team City test status: https://ci.wildfly.org/project.html?projectId=WF_PullRequest&branch_WF_PullRequest=pull%2F18552

File: observability/opentelemetry-api/src/main/java/org/wildfly/extension/opentelemetry/api/WildFlyOpenTelemetryConfig.java [hunk 1]
```diff
@@ -101,13 +49,114 @@ public boolean isMpTelemetryInstalled() {
...
+
+        public Builder setServiceName(String serviceName) {
+            addValue(OTEL_SERVICE_NAME, serviceName);
+            return this;
+        }
+
+        public Builder setSdkDisabled(boolean sdkDisabled) {
+            addValue(OTEL_SDK_DISABLED, "false");
+            return this;
+        }
```
Comment (bstansberry) [inline observability/opentelemetry-api/src/main/java/org/wildfly/extension/opentelemetry/api/WildFlyOpenTelemetryConfig.java:71]: I won't hold merging for this as it's unused, but please file a JIRA to get this removed.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jasondlee. Please note the minor follow-up request I made above.

File: observability/opentelemetry-api/src/main/java/org/wildfly/extension/opentelemetry/api/WildFlyOpenTelemetryConfig.java (hunk 1 above)
Comment (jasondlee) [reply observability/opentelemetry-api/src/main/java/org/wildfly/extension/opentelemetry/api/WildFlyOpenTelemetryConfig.java:71]: I added that as I thought we'd need that in the MP module, but I forgot that, there, we get the props map then override as needed. JIRA: https://issues.redhat.com/browse/WFLY-20158

**Pull request merged**